### PR TITLE
Rename Azure Active Directory to Microsoft Entra ID

### LINKED ISSUE #3009: Microsoft Azure AD to be renamed to Microsoft Entra ID
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://docs.astronomer.io/astro/set-up-scim-provisioning#supported-sso-identity-providers

### The Issue or Context

https://astronomer.slack.com/archives/C015V2JFKT5/p1698106093202829

### Fix or Proposed Change

Rename and keep something like (formerly known as Azure AD) for backward compatibility.

### Additional Notes

_No response_

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #3009 

- Searched docs for `Azure Active Directory` and `Azure AD` and replaced with new name.
- Updated any related links.
- Did not make changes to archived software docs or historical release notes

### CODE REVIEW

File: astro/configure-idp.md [hunk 1]
```diff
@@ -150,29 +150,29 @@ SCIM provisioning allows you to manage Astro users from your identity provider p
...
-This section provides setup steps for setting up Azure AD as your IdP on Astro. After completing this setup, your organization's users can use Azure AD to log in to Astro.
+This section provides setup steps for setting up Microsoft Entra ID as your IdP on Astro. After completing this setup, your organization's users can use Microsoft Entra ID to log in to Astro.
 
 #### Prerequisites
 
 To integrate Azure as your IdP for Astro you must have:
 
 - An Azure subscription.
-- An [Azure AD tenant](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-create-new-tenant) with `Global Administrator` privileges.
+- An [Microsoft Entra ID tenant](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-create-new-tenant) with `Global Administrator` privileges.
```
Comment (lzdanski) [inline astro/configure-idp.md:162]:
```suggestion
- A [Microsoft Entra ID tenant](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-create-new-tenant) with `Global Administrator` privileges.
```

File: astro/install-azure-hybrid.md [hunk 2]
```diff
@@ -25,21 +25,21 @@ To complete the installation, you'll:
...
+- Add the IAM service principal to Microsoft Entra ID that'll be used by Astro.
 
 Astronomer support will create infrastructure within your AWS account to host the resources and Apache Airflow components necessary to deploy DAGs and execute tasks. If you need more than one Astro cluster, contact [Astronomer support](https://cloud.astronomer.io/open-support-request).
 
 ## Prerequisites
 
 - A new [Azure subscription](https://learn.microsoft.com/en-us/dynamics-nav/how-to--sign-up-for-a-microsoft-azure-subscription). For security reasons, Azure subscriptions with existing infrastructure aren't supported. Also, no [Azure policy](https://learn.microsoft.com/en-us/azure/governance/policy/overview) should be applicable to the subscription's [Azure management group](https://docs.microsoft.com/en-us/azure/governance/management-groups/overview).
 
-- An Azure AD user with the following role assignments:
+- An Microsoft Entra ID user with the following role assignments:
```
Comment (lzdanski) [inline astro/install-azure-hybrid.md:36]:
```suggestion
- A Microsoft Entra ID user with the following role assignments:
```

File: learn/airflow-azure-container-instances.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ To complete this tutorial, you need:
 
 ## Step 1: Create an Azure service principal
 
-An Azure service principal is required for external tools like Airflow to connect to your Azure resources. Identify the Azure resource group you want to create your ACI in (or create a new one), then create service principal with write access over that resource group. For more information, see [Use the portal to create an Azure AD application and service principal that can access resources](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal).
+An Azure service principal is required for external tools like Airflow to connect to your Azure resources. Identify the Azure resource group you want to create your ACI in (or create a new one), then create service principal with write access over that resource group. For more information, see [Use the portal to create an Microsoft Entra ID application and service principal that can access resources](https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal).
```
Comment (lzdanski) [inline learn/airflow-azure-container-instances.md:39]:
```suggestion
An Azure service principal is required for external tools like Airflow to connect to your Azure resources. Identify the Azure resource group you want to create your ACI in (or create a new one), then create service principal with write access over that resource group. For more information, see [Use the portal to create a Microsoft Entra ID application and service principal that can access resources](https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal).
```

File: learn/airflow-azure-data-explorer.md [hunk 4]
```diff
@@ -38,7 +38,7 @@ To complete this tutorial, you need:
 
 ## Step 1: Configure your ADX cluster to work with Airflow
 
-To allow Airflow to communicate with your ADX database, you need to configure service principal authentication. To do this, create and register an Azure AD service principal, then give that principal permission to access your ADX database. See [Create an Azure Active Directory application registration in Azure Data Explorer](https://docs.microsoft.com/en-us/azure/data-explorer/provision-azure-ad-app) for more details.
+To allow Airflow to communicate with your ADX database, you need to configure service principal authentication. To do this, create and register an Microsoft Entra ID service principal, then give that principal permission to access your ADX database. See [Create an Microsoft Entra ID application registration in Azure Data Explorer](https://learn.microsoft.com/en-us/azure/data-explorer/provision-entra-id-app) for more details.
```
Comment (lzdanski) [inline learn/airflow-azure-data-explorer.md:41]:
```suggestion
To allow Airflow to communicate with your ADX database, you need to configure service principal authentication. To do this, create and register a Microsoft Entra ID service principal, then give that principal permission to access your ADX database. See [Create a Microsoft Entra ID application registration in Azure Data Explorer](https://learn.microsoft.com/en-us/azure/data-explorer/provision-entra-id-app) for more details.
```

**Review by lzdanski — COMMENTED** [review]
fix grammar

**Review by kentdanas — APPROVED** [review]
Oof the new name is kind of a mouthful, isn't it? This LGTM!

**Review by jwitz — APPROVED** [review]
Looks like you caught all permutations of how we mention Azure AD, so LGTM!

**Pull request merged**